bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/kostalinverter/internal/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,410 @@
+                            valueConfiguration);
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set channel configuration parameters
+        channelConfigs = SecondGenerationChannelConfiguration.getChannelConfiguration();
+        channelConfigsExt = SecondGenerationChannelConfiguration.getChannelConfigurationExt();
+        channelConfigsExtExt = SecondGenerationChannelConfiguration.getChannelConfigurationExtExt();
+        channelConfigsConfigurable = SecondGenerationChannelConfiguration.getChannelConfigurationConfigurable();
+
+        // Set inverter configuration parameters
+        final SecondGenerationInverterConfig inverterConfig = getConfigAs(SecondGenerationInverterConfig.class);
+        this.inverterConfig = inverterConfig;
+
+        // Initialize Gson object gson
+        final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        this.gson = gson;
+
+        // Temporary value during initializing
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Start update as configured
+        secondGenerationPoller = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                refresh();
+                updateStatus(ThingStatus.ONLINE);
+            } catch (RuntimeException scheduleWithFixedDelayException) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        scheduleWithFixedDelayException.getClass().getName() + ":"
+                                + scheduleWithFixedDelayException.getMessage());
+                logger.debug("Error when connecting to inverter, {}: {}", getThing().getUID(),
+                        scheduleWithFixedDelayException.getMessage());
+            } catch (InterruptedException e) {
+                logger.debug("Communication with inverter interrupted, exception {}", e.getMessage());
+            } catch (ExecutionException e) {
+                logger.debug("Communication with inverter failed, exception {}", e.getMessage());
+            } catch (TimeoutException e) {
+                logger.debug("Communication with inverter timed out, exception {}", e.getMessage());
+            }
+        }, 0, SecondGenerationInverterConfig.REFRESHINTERVAL_SEC, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        final ScheduledFuture<?> secondGenerationLocalPoller = secondGenerationPoller;
+
+        if (secondGenerationLocalPoller != null) {
+            secondGenerationLocalPoller.cancel(true);
+            secondGenerationPoller = null;
+        }
+    }
+
+    @SuppressWarnings({ "null", "unchecked" })
+    private void refresh() throws InterruptedException, ExecutionException, TimeoutException {
+        // dxsEntries part
+        String dxsEntriesCall = inverterConfig.url.toString() + "/api/dxs.json?dxsEntries="
```

Comment:
This generates a compiler warning, which is currently suppressed. You could store `inverterConfig` to a local variable and do a null check on that to get rid of the warning.